packages/node_modules/pouchdb-core/src/changes.js

```diff
@@ -169,6 +170,11 @@ Changes.prototype.doChanges = function (opts) {
     opts.filter = '_view';
   }
 
+  /* istanbul ignore if */
+  if (opts.selector && !opts.filter) {
```

Comment:
Good thought. In CouchDB, a selector requires passing `filter=_selector` as a query parameter as well as a `selector` field in the request body. I need to check what CouchDB does if the `filter` value is inconsistent, but we should probably have some validation on the Pouch side to prevent people specifying both.